lib/payment.js

```diff
@@ -126,6 +126,19 @@ class Payment {
         return result;
     }
 
+    /**
+     * Returns the property names which payment hashes are calculated
+     * @return {String[]} Array of property names
+     */
+    static getHashProperties() {
```

Comment:
Why does this have to be a static method? I would think the need for this list is internal to the classes, e.g. a validating function (isSigned()?) should also be a method in these classes.

Reply:
The reason it's static is because other clients may need to hash payments, but without using the Payment class directly. Take for instance the Trading API, it needs to hash payments but it has a different different perspective on payments (validating/processing payments vs making payments). 

This is a different need compared to making a payment, and it has different requirements to the "payment" abstraction, even though they represent the same data structure.

Reply:
I agree with your latest comment, and I think I must have misread your original comment in the Trading API PR where you wanted to have this functionality in the payment class.

And yes, I don't think the API should use this class, at least not for anything else than getting the properties that needs to be part of the hash. This is class for making payments, and the API needs to process payments, which are two separate concerns.

I will instead create a config object, or similar, that exposes the hashing properties. Everyone that needs to hash a payment can use this. Let me know if you have any better ideas.

Reply:
1 - *IF* (and that is a big if) we want to expose the list, then it should probably be in the low-level API (provider) or a new utility class, however
2 - This class is not "for making payments" - it is "a payment" period. Not a receipt, but a payment. (Dont confuse this class with the "Pay" command!) 

In a domain modelled environment such as this, what you do with it is encapsulated in its methods. I still dont really see why the API service would need a different Payment class. The only think missing here is a method for validating a signature, and another Receipts class that could be constructed from a payment and cross signed/verified. A payment is a payment no matter where it is in the API.

NOTE: This is essentially the definition of the TRANSPORT schema - not the STORAGE schema and possible operations and support methods.

Reply:
To put it like this, the Engine **will** be using this class for signature validation